FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -0,0 +1,1457 @@
+{
+    MQTTStatus_t xResult;
+    MQTTConnectInfo_t xConnectInfo;
+    bool xSessionPresent;
+    TransportInterface_t xTransport;
+    MQTTFixedBuffer_t xNetworkBuffer;
+
+    /* Fill the values for network buffer. */
+    xNetworkBuffer.pBuffer = buffer;
+    xNetworkBuffer.size = NETWORK_BUFFER_SIZE;
+
+    /***
+     * For readability, error handling in this function is restricted to the use of
+     * asserts().
+     ***/
+
+    /* Fill in Transport Interface send and receive function pointers. */
+    xTransport.pNetworkContext = pxNetworkContext;
+    xTransport.send = Plaintext_FreeRTOS_send;
+    xTransport.recv = Plaintext_FreeRTOS_recv;
+
+    /* Initialize MQTT library. */
+    xResult = MQTT_Init( pxMQTTContext, &xTransport, prvGetTimeMs, prvSubscriptionManager, &xNetworkBuffer );
+    configASSERT( xResult == MQTTSuccess );
+
+    /* Many fields not used in this demo so start with everything at 0. */
+    memset( ( void * ) &xConnectInfo, 0x00, sizeof( xConnectInfo ) );
+
+    /* Start with a clean session i.e. direct the MQTT broker to discard any
+     * previous session data. Also, establishing a connection with clean session
+     * will ensure that the broker does not store any data when this client
+     * gets disconnected. */
+    xConnectInfo.cleanSession = true;
+
+    /* The client identifier is used to uniquely identify this MQTT client to
+     * the MQTT broker. In a production device the identifier can be something
+     * unique, such as a device serial number. */
+    xConnectInfo.pClientIdentifier = CLIENT_IDENTIFIER;
+    xConnectInfo.clientIdentifierLength = ( uint16_t ) strlen( CLIENT_IDENTIFIER );
+
+    /* Set MQTT keep-alive period. It is the responsibility of the application to ensure
+     * that the interval between Control Packets being sent does not exceed the Keep Alive value.
+     * In the absence of sending any other Control Packets, the Client MUST send a PINGREQ Packet. */
+    xConnectInfo.keepAliveSeconds = MQTT_KEEP_ALIVE_INTERVAL_SECONDS;
+
+    /* Send MQTT CONNECT packet to broker. LWT is not used in this demo, so it
+     * is passed as NULL. */
+    xResult = MQTT_Connect( pxMQTTContext,
+                            &xConnectInfo,
+                            NULL,
+                            CONNACK_RECV_TIMEOUT_MS,
+                            &xSessionPresent );
+
+    if( xResult != MQTTSuccess )
+    {
+        LogError( ( "Connection with MQTT broker failed.\r\n" ) );
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvInitializeCommandContext( CommandContext_t * pxContext )
+{
+    pxContext->complete = false;
+    pxContext->pResponseQueue = NULL;
+    pxContext->returnStatus = MQTTSuccess;
+    pxContext->pPublishInfo = NULL;
+    pxContext->pSubscribeInfo = NULL;
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvDestroyCommandContext( CommandContext_t * pxContext )
+{
+    ( void ) pxContext;
```

Comment:
What is this function for? Is this planned to be having an implementation like this? If so, can we remove this function?

Reply:
I originally had this function since the code used semaphores or condition variables to notify the calling task that the operation was complete. Since I changed it to use task notifications, there weren't any semaphores that needed to be destroyed any more. I kept this here since if the code was changed to use semaphores (e.g. if more than 32 operations were created at a time), then they would need to be destroyed. However, since this function isn't necessary for the current demo, I removed it, and we can just add a comment saying something like "If you decide to use semaphores, they will need to be destroyed"